internal/v2/s2av2.go

```diff
@@ -119,7 +121,7 @@ func (c *s2av2TransportCreds) ClientHandshake(ctx context.Context, serverAuthori
 	}
 	// Remove the port from serverAuthority.
 	serverName := removeServerNamePort(serverAuthority)
-	ctx, cancel := context.WithTimeout(ctx, defaultTimeout)
+	ctx, cancel := context.WithTimeout(ctx, *connectS2ATimeout)
```

Comment:
Do we need the timeout in the HTTP APIs as well?

Reply:
good catch, added thanks